HEATDamage.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Text;
+using System.Diagnostics;
+
+namespace PhysicalDamage.Core
+{
+    class HEATDamage:Damage
+    {
+        public struct HEATData
+        {
+            public float Pierce;
+
+            public float ArmorHEATFactor;
+
+            public float HealthHEATFactor;
+        }
+
+        private HEATData data;
+
+        public HEATDamage(HEATData d,Damage.Target t):base(EDamageTypes.HEAT, t)
+        {
+            data = d;
+        }
+
+        override public float DealDamage()
```

Comment:
After looking through most (I hope all) of your code, I've noticed you don't seem to have a reason to want to use virtual/protected/override but instead an Interface. Interfaces will suite your needs much better. :)